### PR TITLE
azure: Upgrade azidentity to 1.7.0, and deps

### PR DESCRIPTION
**Pull request by benbardin**
[Cloud unit test](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/17789255?hideProblemsFromDependencies=false&hideTestsFromDependencies=false)
[Azure roach tests](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/17789256?hideProblemsFromDependencies=false&hideTestsFromDependencies=false)

Epic: CRDB-42520
Release note: None

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -36,7 +36,7 @@ require (
 	github.com/golang/snappy v0.0.5-0.20231225225746-43d5d4cd4e0e
 	github.com/google/btree v1.0.1
 	github.com/google/pprof v0.0.0-20210827144239-02619b876842
-	github.com/google/uuid v1.5.0 // indirect
+	github.com/google/uuid v1.6.0 // indirect
```
Comment (msbutler) [inline go.mod:39]: this is still core infra :(

**Comment by msbutler** [comment]
also, could you confirm what functionality we're trying to use in the upgraded az dependency?

File: go.mod (hunk 1 above)
Comment (benbardin) [reply go.mod:39]: ...indeed :) put up https://github.com/cockroachdb/cockroach/pull/135324 to upgrade that separately

**Comment by benbardin** [comment]
We want the bugfixes contained in azidentity 1.6.0

**Review by msbutler — APPROVED** [review]

**Comment by benbardin** [comment]
TFTR!

bors r+

**Pull request merged**